### PR TITLE
Use flock to detect the update tool active.

### PR DESCRIPTION
**Pull request by Scron-Chang**
Tested:
Use the 'ls -al' to check the plock file during the update tool
executing.

[root@localhost scron_test]# ls -al /var/run/BIC_FW_updating
-rw-r-x--- 1 root root 0 Apr 14 10:34 /var/run/BIC_FW_updating
[root@localhost scron_test]# ls -al /var/run/BIC_FW_updating
ls: cannot access /var/run/BIC_FW_updating: No such file or directory

The plock file will be removed successfully after updating completely.

Create two ssh clients to do the FW updating. The latter one will be
stopped by the updating tool itself.

Signed-off-by: Scron Chang <Scron.Chang@quantatw.com>

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -703,6 +580,58 @@ static int fw_update(fw_type_t flag, uint8_t *buff, uint32_t buff_len)
...
+    if (fd <= 0)
+    {
+        return -1;
+    }
+
+    return flock(fd, LOCK_EX | LOCK_NB);
+}
+
+/*
+  - Name: lock_plock_file
```
Comment (MouchenHung) [inline src/main.c]: unlock_plock_file

File: src/main.c [hunk 2]
```diff
@@ -703,6 +580,58 @@ static int fw_update(fw_type_t flag, uint8_t *buff, uint32_t buff_len)
...
+  - Description: Unlock the plock_file.
+  - Input:
+      * fd: The file description of plock.
```
Comment (MouchenHung) [inline src/main.c]: file descriptor

File: src/main.c [hunk 3]
```diff
@@ -703,6 +580,58 @@ static int fw_update(fw_type_t flag, uint8_t *buff, uint32_t buff_len)
...
+static int init_process_lock_file(void)
+{
+    return open(plock_file_path, O_RDONLY | O_CREAT);
+}
+
+/*
+  - Name: lock_plock_file
+  - Description: Lock the plock_file.
+  - Input:
+      * fd: The file description of plock.
```
Comment (MouchenHung) [inline src/main.c]: file descriptor

File: src/main.c (hunk 1 above)
Comment (Scron-Chang) [reply src/main.c]: done

**Pull request merged**